### PR TITLE
fix: build 2.3.x from source — use airframe::grammar::grammar_hooks (modify_logits_from_grammar does not exist in any published airframe)

### PR DESCRIPTION
**Pull request by Supersynergy**
## Problem

`shimmy` 2.3.0/2.3.1 cannot be built from source. `src/engine/airframe.rs` calls

```rust
airframe::runtime::gpu::modify_logits_from_grammar(&opts.grammar_mode)
```

but that function does not exist in any published `airframe` release (0.2.9–0.2.11) nor anywhere in airframe's git history — the 2.3.x releases appear to have been built against an unpublished local airframe state. Combined with the stale `airframe-observe` 0.1.0 publish (see Michael-A-Kuykendall/airframe PR: bump airframe-observe to 0.1.1), `cargo install shimmy` fails for everyone, and the 2.3.x GitHub releases carry no binary assets as fallback.

## Fix

Build the grammar mask and control from `airframe::grammar::grammar_hooks` — the API that published airframe 0.2.11 actually exports. Per its docs both hooks share one `GrammarState` (the pre-sample mask reads it, the post-sample control advances it), so when a grammar mode is active its control takes precedence over the FSE reject-pattern control; with `grammar_mode: "none"` behavior is unchanged (both hooks `None`).

Diff is minimal (17+/7-, CRLF preserved).

## Verified

macOS arm64, rustc 1.97.1, Metal:
- builds against crates.io `airframe` 0.2.11 (with `airframe-observe` patched to git, pending its 0.1.1 publish)
- `shimmy list` auto-discovers 31 local GGUF models (Ollama + LM Studio dirs)
- `shimmy generate … --max-tokens 8` streams tokens through the Airframe/Metal pipeline

### CODE REVIEW

**Comment by Michael-A-Kuykendall** [comment]
Thanks for the diagnosis and the fix proposal. Tracking in issue #212; integrating via shimmy 2.3.2 through our own release process. Closing per CONTRIBUTING.md (maintainer-only PRs). Email michaelallenkuykendall@gmail.com if you'd like to contribute.
